### PR TITLE
Update input styles to improve accessibility for interactive components

### PR DESCRIPTION
**Pull request by noranda**
## What did we change?
- Updates focus outline style for all interactive components
- Updates `EzCheckbox` styles and adds hover state to super checkbox buttons (see below)
- Updates `EzRadio` styles (see below)
- Updates `EzField` input and select styles (see below)
- Adds additional colors to stitches theme to support design: `gray250: '#f5f7f7'`, `gray350: '#e5e6e6'`, and `gray650: '#787878'`
- Updates emotion `common.disabled` color to `neutral140`
- Adds `iconColor` and `hover` options for super radio and super checkbox button custom themes
- Replaces custom check and chevron icons with FontAwesome icons
- Replaces `EzTable` icons with `EzIconButton`
- Adds alerts to doc page for `EzBadge`, `EzButton`, `EzCheckbox`, `EzChip`, `EzIcon`, `EzIconButton`, `EzRadio`, and `EzRating` color sections to clarify that examples only show a subset of colors available
- Updates tests and types as needed

## Why are we doing this?
Designer [request](https://www.figma.com/file/sVDwACKoMGvES1JbX6AuSW/Ecomm-Core-Screens?node-id=0-1&t=tsuNySb82wJnevFT-0) to approve accessibility.

### Focus outlines:
Screenshot 2023-05-02 at 1 32 57 PM Screenshot 2023-05-02 at 1 33 54 PM Screenshot 2023-05-02 at 1 34 41 PM Screenshot 2023-05-02 at 1 36 07 PM Screenshot 2023-05-02 at 1 37 01 PM Screenshot 2023-05-02 at 1 37 24 PM Screenshot 2023-05-02 at 1 37 56 PM Screenshot 2023-05-02 at 1 38 35 PM
Screenshot 2023-05-02 at 1 39 14 PM

### `EzCheckbox` style updates:
Old: Screenshot 2023-05-02 at 1 41 36 PM Screenshot 2023-05-02 at 1 40 33 PM
New: Screenshot 2023-05-02 at 1 41 59 PM Screenshot 2023-05-02 at 1 41 03 PM
Super checkbox buttons old:
Screenshot 2023-05-02 at 1 43 27 PM
Super checkbox buttons new:
Screenshot 2023-05-02 at 1 44 32 PM
Disabled old:
Screenshot 2023-05-02 at 1 46 24 PM
Disabled new:
Screenshot 2023-05-02 at 1 47 11 PM

### `EzRadio` style updates:
Old: Screenshot 2023-05-02 at 1 48 18 PM
New: Screenshot 2023-05-02 at 1 49 06 PM
Super radio buttons old:
Screenshot 2023-05-02 at 1 49 44 PM
Super radio buttons new:
Screenshot 2023-05-02 at 1 50 04 PM
Disabled old:
Screenshot 2023-05-02 at 1 50 24 PM
Disabled new:
Screenshot 2023-05-02 at 1 50 47 PM

### `EzField` style updates:
Search old:
Screenshot 2023-05-02 at 1 52 36 PM Screenshot 2023-05-02 at 1 56 32 PM Screenshot 2023-05-02 at 1 55 04 PM
Search new:
Screenshot 2023-05-02 at 1 53 02 PM Screenshot 2023-05-02 at 1 53 57 PM Screenshot 2023-05-02 at 1 55 36 PM
Select old:
Screenshot 2023-05-02 at 1 58 14 PM
Screenshot 2023-05-02 at 2 05 44 PM
Screenshot 2023-05-02 at 2 06 11 PM
Select new:
Screenshot 2023-05-02 at 2 06 55 PM
Screenshot 2023-05-02 at 2 07 07 PM
Screenshot 2023-05-02 at 2 07 20 PM

## Checklist

- [x] Provide test coverage for the changes made
- [x] Create a changeset (`yarn changeset`) with a summary of the changes

### CODE REVIEW

File: .changeset/rare-files-exist.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+'@ezcater/recipe': patch
```
Comment (pgrippi) [inline .changeset/rare-files-exist.md:2]: this seems large enough that it probably warrants a minor release

**Review by pgrippi — DISMISSED** [review]
These styles are getting complex! 😵‍💫

It makes me want to see if there is a better way to integrate our overrides with MUI given how many styles you have to pass to `sx` props, but lets save that as an action item for the future!

File: .changeset/rare-files-exist.md (hunk 1 above)
Comment (noranda) [reply .changeset/rare-files-exist.md:2]: Good point! I'll add a minor changeset.

File: packages/recipe/src/components/EzField/EzAutosuggest.tsx [hunk 2]
```diff
@@ -31,7 +32,9 @@ const EzAutosuggest = props => {
       <TextInputWrapper className={props.className} disabled={props.disabled}>
         <EzTextInput {...inputProps} ref={triggerRef} error={props.error} touched={props.touched} />
         <InsetIcon insetY0 right0 pr2>
-          <ChevronIcon flip={state.isOpen} />
+          <div style={{fontSize: '13px'}}>
```
Comment (danidewitt) [inline packages/recipe/src/components/EzField/EzAutosuggest.tsx:35]: do we not have an icon size that works here? does it have to be an inline style?

File: packages/recipe/src/components/EzField/EzDateInput.tsx [hunk 3]
```diff
@@ -129,7 +130,9 @@ const EzDateInput = ({
         </InsetIcon>
         <EzTextInput {...comboboxInput} type="text" />
         <InsetIcon insetY0 right0 pr2>
-          <ChevronIcon flip={isOpen} />
+          <div style={{fontSize: '13px'}}>
```
Comment (danidewitt) [inline packages/recipe/src/components/EzField/EzDateInput.tsx:133]: same question

File: packages/recipe/src/components/EzField/EzSelect.tsx [hunk 4]
```diff
@@ -67,7 +69,9 @@ const EzSelect = props => {
           touched={props.touched}
         />
         <InsetIcon insetY0 right0 pr2>
-          <ChevronIcon flip={state.isOpen} />
+          <div style={{fontSize: '13px'}}>
```
Comment (danidewitt) [inline packages/recipe/src/components/EzField/EzSelect.tsx:72]: same question

**Review by danidewitt — COMMENTED** [review]
just the same question 3 across 3 components - because inline styles in our components cause unnecessary re-renders in downstream apps

File: packages/recipe/src/components/EzField/EzDateInput.tsx [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
 import {TextInputWrapper} from './Picker.styles';
-import EzPopover from '../EzPopover';
 import EzCalendar from '../EzCalendar/EzCalendar';
 import {useMenuTrigger, useMenuTriggerState} from '../Overlays';
 import {useUpdateEffect} from '../../utils/hooks';
-import {ChevronIcon, CalendarIcon, InsetIcon} from '../Icons';
+import {CalendarIcon, InsetIcon} from '../Icons';
 import FocusScope from '../utils/FocusScope';
 import {clsx} from '../../utils';
+import {EzIcon, EzPopover} from '..';
```
Comment (pgrippi) [inline packages/recipe/src/components/EzField/EzDateInput.tsx:13]: same here with the circular dependency

File: packages/recipe/src/components/EzField/EzAutosuggest.tsx [hunk 6]
```diff
@@ -1,11 +1,12 @@
...
+import {faChevronDown, faChevronUp} from '@fortawesome/free-solid-svg-icons';
 import {TextInputWrapper} from './Picker.styles';
 import theme from '../theme.config';
 import EzTextInput from './EzTextInput';
-import EzPopover from '../EzPopover';
-import {ChevronIcon, InsetIcon} from '../Icons';
+import {InsetIcon} from '../Icons';
 import EzListBox from './EzListBox';
 import {useComboBoxState, useComboBox} from './useComboBox';
+import {EzIcon, EzPopover} from '..';
```
[the remark this answers is not in the document]
Comment (noranda) [reply packages/recipe/src/components/EzField/EzAutosuggest.tsx:9]: Oops, autofix fail! Will fix.

File: packages/recipe/src/components/EzField/EzListBox.tsx [hunk 7]
```diff
@@ -1,18 +1,19 @@
 import React, {useRef} from 'react';
 import Highlighter from 'react-highlight-words';
+import {faCheck} from '@fortawesome/free-solid-svg-icons';
+import {Stack} from '@mui/material';
 import theme from '../theme.config';
 import {useUniqueId, useScrollIntoView} from '../../utils/hooks';
 import {clsx} from '../../utils';
+import {EzIcon} from '..';
```
Comment (danidewitt) [inline packages/recipe/src/components/EzField/EzListBox.tsx:8]: i noticed this one too - maybe you're fixing it with peter's other comments

File: packages/recipe/src/components/EzField/EzSelect.tsx [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
 import theme from '../theme.config';
 import {TextInputWrapper} from './Picker.styles';
 import EzTextInput from './EzTextInput';
 import EzPopover from '../EzPopover';
-import {ChevronIcon, InsetIcon} from '../Icons';
+import {InsetIcon} from '../Icons';
 import EzListBox from './EzListBox';
 import {useSelectState, useSelect} from './useSelect';
 import {domProps} from '../../utils';
+import {EzIcon} from '..';
```
Comment (danidewitt) [inline packages/recipe/src/components/EzField/EzSelect.tsx:11]: and this one

File: packages/recipe/src/components/EzField/EzListBox.tsx (hunk 7 above)
Comment (noranda) [reply packages/recipe/src/components/EzField/EzListBox.tsx:8]: Yeah actually there are a bunch of them including files I didn't touch. I fixed them all just to be thorough!

**Review by pgrippi — APPROVED** [review]

**Review by danidewitt — APPROVED** [review]

**Pull request merged**